README.md

```diff
@@ -102,154 +157,331 @@ jobs:
+on: [push]
+
+name: Run Azure Login With a Service Principal Secret
+
+jobs:
+
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    steps:
+    
+    - uses: azure/login@v1
       with:
-        creds: ${{ secrets.AZURE_US_GOV_CREDENTIALS }}
-        environment: 'AzureUSGovernment'
-        enable-AzPSSession: false
-    - name: Login to Azure US Gov Cloud with Az Powershell
-      uses: azure/login@v1
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+    
+    - name: Azure CLI script
+      uses: azure/CLI@v1
+      env:
+        AZURE_CORE_OUTPUT: none
       with:
-        creds: ${{ secrets.AZURE_US_GOV_CREDENTIALS }}
-        environment: 'AzureUSGovernment'
-        enable-AzPSSession: true
-```
+        azcliversion: latest
+        inlineScript: |
+          az account show
 
-Refer to the [Azure PowerShell](https://github.com/azure/powershell) GitHub Action to run your Azure PowerShell scripts.
+```
 
-## Sample Azure Login workflow that uses Azure login action to run az cli on Azure Stack Hub
+- **The worklfow sample to run both Azure CLI and Azure PowerShell**
 
 ```yaml
 # File: .github/workflows/workflow.yml
 
 on: [push]
 
-name: AzureLoginSample
+name: Run Azure Login With a Service Principal Secret
 
 jobs:
 
   build-and-deploy:
     runs-on: ubuntu-latest
     steps:
+    
     - uses: azure/login@v1
       with:
         creds: ${{ secrets.AZURE_CREDENTIALS }}
-        environment: 'AzureStack'
+        enable-AzPSSession: true
+    
+    - name: Azure CLI script
+      uses: azure/CLI@v1
+      env:
+        AZURE_CORE_OUTPUT: none
+      with:
+        azcliversion: latest
+        inlineScript: |
+          az account show
+
+    - name: Azure PowerShell script
+      uses: azure/powershell@v1.2.0
+      with:
+        azPSVersion: "latest"
+        inlineScript: |
+          Get-AzWebApp
+```
 
-    - run: |
-        az webapp list --query "[?state=='Running']"
+If you want to pass subscription ID, tenant ID, client ID, and client secret as individual parameters instead of bundling them in a single JSON object to address the [security concerns](https://docs.github.com/actions/security-guides/encrypted-secrets), below snippet can help with the same.
 
+```yaml
+  - uses: Azure/login@v1
+    with:
+      creds: '{"clientId":"${{ secrets.AZURE_CLIENT_ID }}","clientSecret":"${{ secrets.AZURE_CLIENT_SECRET }}","subscriptionId":"${{ secrets.AZURE_SUBSCRIPTION_ID }}","tenantId":"${{ secrets.AZURE_TENANT_ID }}"}'
 ```
 
-Refer to the [Azure Stack Hub Login Action Tutorial](https://learn.microsoft.com/azure-stack/user/ci-cd-github-action-login-cli?view=azs-2008) for more detailed instructions.
+### Login With System-assigned Managed Identity
 
-## Configure deployment credentials
-  
-### Configure a service principal with a secret
+> [!NOTE]
+>
+> "Login With System-assigned Managed Identity" is only supported on GitHub self-hosted runners and the self-hosted runners need to be hosted by Azure virtual machines.
+
```

Comment:
```suggestion
> [!WARNING]
> Avoid using managed identity login on self-hosted runners in public repositories. Managed identities enable secure authentication with Azure resources and obtain Azure AD tokens without the need for explicit credential management. Any user can open pull requests against your repository and access your self-hosted runners without credentials. See more details in [self-hosted runner security](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners#self-hosted-runner-security).
```
@jiasli, @Jacekey23, shall we add a warning message for using manage identity on self-hosted runners in public repositories?

Reply:
cc @dcaro @yonzhan

Reply:
We should have this message at the beginning of this article.

Reply:
@dcaro, @MoChilia  Is the warning message content OK now?

Reply:
It's good for me. Agree to have this message at the beginning.